server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeValuesCollectorQueue.java

```diff
@@ -230,13 +232,23 @@ LeafBucketCollector getLeafCollector(Comparable forceLeadSourceValue,
                                          LeafReaderContext context, LeafBucketCollector in) throws IOException {
         int last = arrays.length - 1;
         LeafBucketCollector collector = in;
+        boolean requiresRehashing = false;
```

Comment:
We call this function multiple times per leaf in the sorted case. So the remapping is not always necessary. We should compare the context.ord with the previous one to avoid doing the work when it's not needed.

Reply:
good idea. addressed in 64b1872